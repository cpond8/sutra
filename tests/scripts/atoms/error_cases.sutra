;; error_cases.sutra
;; Test error handling and edge cases for atoms

;; Division by zero error
(print "Testing division by zero:")
;; This should cause an error, so we'll comment it out for now
;; (print (/ 10 0))

;; Type errors - trying to do arithmetic on non-numbers
(print "Testing type errors:")
;; (print (+ "hello" 5))

;; Arity errors - wrong number of arguments
(print "Testing arity errors:")
;; (print (+))  ;; Too few arguments
;; (print (- 10 5 2))  ;; Too many for subtraction

;; Test error atom
(print "Testing explicit error:")
;; (error "This is a test error")

;; For now, let's test valid edge cases
(print "Valid edge cases:")
(print (+ 0 0))
(print (- 0 0))
(print (* 0 5))
(print (/ 0 1))
(print (mod 5 1))
(print (eq? 0 0))
(print (not false))

;; Test with negative numbers
(print "Negative number operations:")
(print (+ -5 -3))
(print (- -10 -5))
(print (* -2 -3))
(print (/ -12 -3))
(print (mod -7 3))
