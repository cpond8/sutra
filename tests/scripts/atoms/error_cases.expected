Testing division by zero:
Testing type errors:
Testing arity errors:
Testing explicit error:
Valid edge cases:
0
0
0
0
0
true
true
Negative number operations:
-8
-5
6
4
-1
